File: app/src/main/java/com/xuie/imaginaryandroid/data/source/NETSSource.java
package com.xuie.imaginaryandroid.data.source;

import com.xuie.imaginaryandroid.data.NetsDetail;
import com.xuie.imaginaryandroid.data.NetsSummary;

import java.util.List;

import rx.Observable;

/**
 * Created by xuie on 17-8-17.
 */

public interface NETSSource {
    Observable<List<NetsSummary>> getNews(int page);

    Observable<NetsDetail> getNewDetail(String postId);
}

File: app-mvp-kotlin/src/main/java/com/xuie/imaginaryandroid/data/source/NETSSource.kt
package com.xuie.imaginaryandroid.data.source

import com.xuie.imaginaryandroid.data.NetsDetail
import com.xuie.imaginaryandroid.data.NetsSummary
import com.xuie.imaginaryandroid.data.VideoBean
import io.reactivex.Observable
import io.reactivex.Single

/**
 * Created by xuie on 17-8-17.
 */

interface NETSSource {
    fun getNews(page: Int): Single<MutableList<NetsSummary>>?

    fun getNewDetail(postId: String): Observable<NetsDetail>

    fun getVideoList(type: String, page: Int): Single<MutableList<VideoBean>>?
}
